src/test/scala/org/broadinstitute/hail/methods/SkatoSuite.scala

```diff
@@ -0,0 +1,387 @@
+              (groupName, (pValue, nMarkers, nMarkersTested))}
+            }.toArray
+          })
+
+          // Annotate samples with covariates and phenotype
+          writeTextFile(sampleAnnotations, sc.hadoopConfiguration) { w =>
+            val sb = new StringBuilder()
+            sb.append("Sample\tPhenotype")
+            for (j <- 0 until nCovar) {
+              sb.append(s"\tC$j")
+            }
+            sb.append("\n")
+            w.write(sb.result())
+
+            s.vds.sampleIds.zipWithIndex.foreach { case (s, i) =>
+              val sb = new StringBuilder()
+              sb.append(s)
+              sb.append("\t")
+              sb.append(y(i).getOrElse("NA"))
+
+              for (j <- 0 until nCovar) {
+                sb.append("\t")
+                sb.append(x(j)(i).getOrElse("NA"))
+              }
+              sb.append("\n")
+              w.write(sb.result())
+            }
+          }
+
+          val sb = new StringBuilder()
+          if (!quantitative)
+            sb.append("Phenotype: Boolean")
+          else
+            sb.append("Phenotype: Double")
+
+          for (j <- 0 until nCovar) {
+            sb.append(s",C$j: Double")
+          }
+          s = AnnotateSamplesTable.run(s, Array("-i", sampleAnnotations, "-r", "sa.pheno", "-t", sb.result()))
+
+          // Annotate variants with group names
+          writeTextFile(variantAnnotations, sc.hadoopConfiguration) { w =>
+            w.write("Variant\tGroup\n")
+            groups.foreach { case (v, i) => w.write(v.toString + "\t" + s"group_$i\n")
+            }
+          }
+          s = AnnotateVariants.run(s, Array("table", variantAnnotations, "-r", "va.groups", "-v", "Variant"))
+
+          // Run SKAT-O command from Hail
+          writeConfig(configSkato)
+
+          var cmd = collection.mutable.ArrayBuffer("-k", "va.groups.Group", "--block-size", blockSize.toString,
+            "-y", "sa.pheno.Phenotype", "--config", configSkato, "-o", hailOutputFile, "--seed", seed.toString,
+            "--kernel", kernel, "--n-resampling", nResampling.toString, "--type-resampling", typeResampling, "--impute-method", imputeMethod,
+            "--method", method, "--r-corr", rCorr.mkString(","), "--missing-cutoff", missingCutoff.toString,
+            "--weights-beta", weightsBeta.mkString(","), "--estimate-maf", estimateMAF.toString
+          )
+
+          if (quantitative)
+            cmd += "-q"
+
+          if (nCovar > 0) {
+            cmd += "-c"
+            cmd += (0 until nCovar).map{i => s"sa.pheno.C$i"}.mkString(",")
+          }
+
+          if (noAdjustment)
+            cmd += "--no-adjustment"
+
+          if (!(rCorr.length == 1 && rCorr(0) == 0.0) && kernel != "linear" && kernel != "linear.weighted"){
+            try {
+              s = GroupTestSKATO.run(s, cmd.toArray)
+              false
+            } catch {
+              case e: FatalException => true
```

Comment:
You can replace this with `expect[FatalException]` or `expectFatal` if you want to verify the fatal message matches a regex.